src/js/actions/ContextIdActions.js

```diff
@@ -56,7 +56,11 @@ export const changeCurrentContextId = contextId => {
       const dirty = await repo.isDirty(['/contextId.json', '/.categories']);
```

Comment:
This is going to produce some slowdown still since it's using `git.status`. I don't know if there's anyway around that so maybe we should just always attempt a save?
This will produce a lot of noise in the git log, but it may be worth the performance boost.